lib/metrics.py

```diff
@@ -67,9 +67,14 @@ def emit(self, stats):
             self.fallback_emitter.emit(stats)
 
 
-class MetricsEmitterThread(threading.Thread):
+class BaseMetricsEmitterThread(threading.Thread, metaclass=ABCMeta):
+    """
+    This base class contains all boilerplate code needed to emit metrics.
+    Only thing left to implement in subclass is `_selected_stats_to_emit` method.
```

Comment:
Could you update this dosctring? Since `_selected_stats_to_emit` is a property now and there is another abstract method `_gather_metrics`.

Reply:
Please check now.